shell/src/main/java/org/apache/accumulo/shell/commands/ConfigCommand.java

```diff
@@ -127,19 +127,29 @@ public int execute(final String fullCommand, final CommandLine cl, final Shell s
       value = pair[1];
 
       if (tableName != null) {
-        if (!Property.isValidTablePropertyKey(property)) {
-          throw new BadArgumentException("Invalid per-table property.", fullCommand,
-              fullCommand.indexOf(property));
+        if (!Property.isTablePropertyValid(property, value)) {
+          if (!Property.isValidTablePropertyKey(property)) {
+            throw new BadArgumentException("Invalid per-table property.", fullCommand,
+                fullCommand.indexOf(property));
+          } else {
+            throw new BadArgumentException("Invalid property value.", fullCommand,
+                fullCommand.indexOf(value));
+          }
         }
         if (property.equals(Property.TABLE_DEFAULT_SCANTIME_VISIBILITY.getKey())) {
           new ColumnVisibility(value); // validate that it is a valid expression
         }
         shellState.getAccumuloClient().tableOperations().setProperty(tableName, property, value);
         Shell.log.debug("Successfully set table configuration option.");
```

Comment:
It looks like a lot of this code is duplicated in the next `else if` block. I wonder if it would be worth it to extract some of the logic into a new method for reuse.

Reply:
Yes, extracting this out to another method cleaned up a good bit of the if/else logic in the "set operation" block.